src/Esfa.Vacancy.Infrastructure/Services/ApprenticeshipSearchService.cs

```diff
@@ -42,92 +39,106 @@ public ApprenticeshipSearchService(IProvideSettings provideSettings, ILog logger
-                            break;
-                    }
-
-                    return search;
-                }).ConfigureAwait(false);
-
-                _logger.Info($"Retrieved {esReponse.Total} apprenticeships from Elastic search with parameters {parameters}");
+                searchClientResponse = _apprenticeshipSearchClient.Search(searchClientParameters);
             }
-            catch (WebException e)
+            catch (Exception e)
             {
                 throw new InfrastructureException(e);
             }
+            
+            var searchResponse = new SearchApprenticeshipVacanciesResponse()
+            {
+                TotalMatched = searchClientResponse.Total,
+                TotalReturned = searchClientResponse.Results.Count(),
+                TotalPages = searchClientResponse.TotalPages,
+                CurrentPage = parameters.PageNumber,
+                SortBy = parameters.SortBy,
+                ApprenticeshipSummaries = searchClientResponse.Results.Select(GetApprenticeshipSummary)
+            };
+
+            return searchResponse;
+        }
 
-            if (!esReponse.ConnectionStatus.Success)
+        private ApprenticeshipSearchRequestParameters GetSearchClientParameters(VacancySearchParameters parameters)
+        {
+            var searchClientParameters = new ApprenticeshipSearchRequestParameters
             {
-                var ex = new Exception("Unexpected response received from Elastic Search");
-                throw new InfrastructureException(ex);
-            }
+                FrameworkLarsCodes = parameters.FrameworkLarsCodes,
+                StandardLarsCodes = parameters.StandardLarsCodes,
+                PageSize = parameters.PageSize,
+                PageNumber = parameters.PageNumber,
+                FromDate = parameters.FromDate,
+                VacancyLocationType = string.IsNullOrEmpty(parameters.LocationType) ? VacancyLocationType.Unknown : (VacancyLocationType)Enum.Parse(typeof(VacancyLocationType), parameters.LocationType),
+                Longitude = parameters.Longitude,
+                Latitude = parameters.Latitude,
+                SearchRadius = parameters.DistanceInMiles,
+                CalculateSubCategoryAggregations = false
+            };
 
-            if (parameters.HasGeoSearchFields)
+            switch (parameters.SortBy)
             {
-                _distanceSetter.SetDistance(parameters, esReponse);
+                case SortBy.Age:
+                    searchClientParameters.SortType = VacancySearchSortType.RecentlyAdded;
+                    break;
+                case SortBy.ExpectedStartDate:
+                    searchClientParameters.SortType = VacancySearchSortType.ExpectedStartDate;
+                    break;
+                case SortBy.Distance:
+                    searchClientParameters.SortType = VacancySearchSortType.Distance;
+                    break;
+                default:
```

Comment:
Sort by Age includes sorting by recentlyadded and distance... this was also the default sort order. I am not sure relevancy is same as this.. but it if is not then this has changed the behavior.

Reply:
Thanks Deven. In the client a sort type of RecentlyAdded
sorts by Posted Date, then Distance, then vacancy reference number.  (The addition of sort by Distance has been added so we return the GeoDistance data if applicable)

https://github.com/SkillsFundingAgency/das-shared-packages/blob/37e5fc8b88d312147cd4177eb9355befe97f4361/SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/ApprenticeshipSearchClient.cs#L225

The parameters.SortBy is always explcitly set so default should never be called.  I've just added a default to be more explicit.

https://github.com/SkillsFundingAgency/vacancy-register-api/blob/master/src/Esfa.Vacancy.Application/Queries/SearchApprenticeshipVacancies/SortByCalculator.cs